### PR TITLE
[Calling][Feature] Join Teams meeting with id and passcode

### PR DESCRIPTION
**Pull request by ShaunaSong**
## Purpose

* Add the locator to join the teams meeting using id and passcode

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java [hunk 1]
```diff
@@ -703,6 +706,10 @@ private void launchComposite(final Context context,
         if (locator instanceof CallCompositeGroupCallLocator) {
             callType = CallType.GROUP_CALL;
             groupId = ((CallCompositeGroupCallLocator) locator).getGroupId();
+        } else if (locator instanceof CallCompositeTeamsMeetingIdLocator) {
+            callType = CallType.TEAMS_MEETING;
+            meetingId = ((CallCompositeTeamsMeetingIdLocator) locator).getMeetingId();
```
Comment (ahammer) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java:711]: Cast once to a local variable.

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java [hunk 2]
```diff
@@ -772,6 +783,10 @@ private void launchComposite(final Context context,
...
```
Comment (ahammer) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java:788]: same, cast once to a local variable

File: azure-communication-ui/demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/CallLauncherActivity.kt [hunk 3]
```diff
@@ -362,10 +368,14 @@ class CallLauncherActivity : AppCompatActivity() {
...
+        var meetingId: String? = null
+        var meetingPasscode: String? = null
         if (binding.teamsMeetingRadioButton.isChecked) {
             meetingLink = binding.groupIdOrTeamsMeetingLinkText.text.toString()
-            if (meetingLink.isBlank()) {
-                val message = "Teams meeting link is invalid or empty."
+            meetingId = binding.teamsMeetingId.text.toString()
+            meetingPasscode = binding.teamsMeetingPasscode.text.toString()
+            if (meetingId.isBlank() && meetingLink.isBlank()) {
+                val message = "Teams meeting Id and meeting link are invalid or empty."
```
Comment (ahammer) [inline azure-communication-ui/demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/CallLauncherActivity.kt:378]: Should string be a resource? Localized?

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt [hunk 4]
```diff
@@ -445,7 +448,18 @@ internal class CallingSDKWrapper(
...
+                CallType.TEAMS_MEETING -> {
+                    if (!callConfig.meetingLink.isNullOrEmpty()) {
+                        TeamsMeetingLinkLocator(callConfig.meetingLink)
+                    } /* <MEETING_ID_LOCATOR> */ else if (!callConfig.meetingId.isNullOrEmpty() && !callConfig.meetingPasscode.isNullOrEmpty()) {
+                        TeamsMeetingIdLocator(callConfig.meetingId, callConfig.meetingPasscode)
+                    } /* </MEETING_ID_LOCATOR> */ else {
+                        throw CallCompositeException(
+                            "Teams Meeting information is incomplete",
+                            IllegalStateException()
+                        )
```
Comment (iaulakh) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt:460]: Please check if we need to return error code call join failed, or may be handled in middleware?

**Review by iaulakh — DISMISSED** [review]
@ShaunaSong It will be great if you can write one end to end Ui test for call join with mocking. You can also add in KTLO bucket where we have all pending UI tests to write. (once pipelines fixed)

**Review by ahammer — DISMISSED** [review]

**Review by iaulakh — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Pull request merged**